sympy/combinatorics/perm_groups.py

```diff
@@ -5075,3 +5075,132 @@ def _stabilizer(degree, generators, alpha):
     return [_af_new(x) for x in stab_gens]
 
 PermGroup = PermutationGroup
+
+class SymmetricPermutationGroup(Basic):
+    """
+    The class defining the lazy form of SymmetricGroup
+
+    deg : int
+    """
+
+    def __new__(cls, deg):
+        deg = sympify(deg)
+        obj = Basic.__new__(cls, deg)
+        obj.deg = deg
+        obj._order = None
```

Comment:
I don't think that this have to be cached yet. It can be still be time consuming to compute the factorials, but even naively computing factorials every time can be better than `PermutationGroup`

Reply:
Okay, I already see PermutationGroup caches order.

Reply:
So should I leave it as it is?

Reply:
Yes

Reply:
sure